backend/tests/conftest.py

```diff
@@ -1,65 +1,181 @@
+            fetched = (await sess.execute(query)).scalars().all()
+        return [obj for obj in fetched]
+
+    return _helper
+
+
+@fixture
+async def update_object(get_session: typing.AsyncGenerator[AsyncSession, None]):
+    """
+    A fixture that provides a helper method that perform a database update
+    for the object passed as the argument, into the specified table
+    """
+
+    async def _helper(
+        object: typing.List[sqlalchemy.Table],
+        id: int,
+        table: sqlalchemy.Table,
+        subquery_relation: InstrumentedAttribute = None,
+    ):
+        async with get_session() as sess:
+            query = await sess.execute(select(table).filter(table.id == id))
+            db_obj = query.scalar_one_or_none()
+
+            for field, value in object:
+                if value is not None:
+                    setattr(db_obj, field, value)
+            await sess.flush()
+        await sess.refresh(db_obj)
+        return db_obj
+
+    return _helper
+
+
+@fixture
+def read_objects(get_session: typing.AsyncGenerator[AsyncSession, None]):
+    """
+    A fixture that provides a helper method that perform a database
+    read all operation using the specified statement.
+    """
+
+    async def _helper(
+        stmt,
+    ):
+        async with get_session() as sess:
+            fetched = (await sess.execute(stmt)).scalars().all()
+        return [obj for obj in fetched]
+
+    return _helper
+
+
+@fixture
+def read_object(get_session: typing.AsyncGenerator[AsyncSession, None]):
+    """
+    A fixture that provides a helper method that perform a database
+    read operation using the specified statement.
+    """
+
+    async def _helper(
+        stmt,
+    ):
+        async with get_session() as sess:
+            return (await sess.execute(stmt)).scalars().one_or_none()
 
     return _helper
 
 
 @fixture(autouse=True)
-async def startup_event_force():
-    async with LifespanManager(backend_app):
-        yield
+async def clean_up_database(get_session: typing.AsyncGenerator[AsyncSession, None]) -> None:
+    """Clean up the database after test run."""
+    yield
+    async with get_session() as sess:
+        await sess.execute(sqlalchemy.text(f"""DELETE FROM {models.Booking.__tablename__};"""))
+        await sess.execute(sqlalchemy.text(f"""DELETE FROM {models.Job.__tablename__};"""))
+        await sess.execute(sqlalchemy.text(f"""DELETE FROM {models.Inventory.__tablename__};"""))
+        await sess.execute(sqlalchemy.text(f"""DELETE FROM {models.Feature.__tablename__};"""))
+        await sess.execute(sqlalchemy.text(f"""DELETE FROM {models.Product.__tablename__};"""))
+        await sess.execute(sqlalchemy.text(f"""DELETE FROM {models.LicenseServer.__tablename__};"""))
+        await sess.execute(sqlalchemy.text(f"""DELETE FROM {models.Configuration.__tablename__};"""))
+        await sess.execute(sqlalchemy.text(f"""DELETE FROM {models.Cluster.__tablename__};"""))
+        await sess.commit()
```

Comment:
This is going to make your tests very slow. The way to handle this is to make the session during unit tests to flush their objects instead of commit. The database should be rolled back after each test function finishes.

Reply:
We may revisit this in the future, but resolving for now.